pkg/build/api/v1/types.go

```diff
@@ -42,6 +42,9 @@ type BuildSpec struct {
 
 	// Compute resource requirements to execute the build
 	Resources kapi.ResourceRequirements `json:"resources,omitempty" description:"the desired compute resources the build should have"`
+
+	// Secrets is a list of additional secrets that will be included in the build pod
+	Secrets []SecretSpec `json:"secrets,omitempty" description:"a list of secrets to include in the build pod in addition to pull, push and source secrets"`
```

Comment:
if it's something that can be applied to any build pod, I didn't think it was worth replicating in each strategy spec, but I can see we'd want to be consistent with other stuff